### PR TITLE
feat(model): Add model hydra yaml

### PR DESCRIPTION
**Pull request by boat1603**
## Why this PR
Why we need this PR?
This PR is about adding auto download model feature from hydra yaml config.

## Changes
- Add falcon model example config.

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/model/configuration_example/model/falcon-7b-from-pretrained.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  use_cache: true
+  apply_residual_connection_post_layernorm: false
+  hidden_dropout: 0.0
+  attention_dropout: 0.0
+  multi_query: true
+  alibi: false
+  bias: false
+  parallel_attn: true
+tokenizer:
+  pretrained: tiiuae/falcon-7b
```
Comment (new5558) [inline src/model/configuration_example/model/falcon-7b-from-pretrained.yaml:18]: pretrained is a little too specific that we need to load the tokenizer from huggingface hub. I think we should have more generic name like 'tokenizer_name'

File: src/model/openthaigpt_pretraining_model/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from .constants import TOKENIZERS, MODELS, MODEL_CONFIGS
+
+
+def load_model(model_config):
```
Comment (new5558) [inline src/model/openthaigpt_pretraining_model/models/__init__.py:4]: Can we have load_model and load_tokenizer functions for simplicity?

Comment (new5558) [reply src/model/openthaigpt_pretraining_model/models/__init__.py:4]: No need to refactor config file

File: src/model/openthaigpt_pretraining_model/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    tokenizer_object = TOKENIZERS.get(model_config.tokenizer.tokenizer_class, None)
+    if tokenizer_object is None:
+        raise NotImplementedError(
+            f"No tokenizer name: {model_config.tokenizer.tokenizer_class}"
+        )
+    model_config_object = MODEL_CONFIGS.get(model_config.name, None)
```
Comment (new5558) [inline src/model/openthaigpt_pretraining_model/models/__init__.py:10]: Can load_model handle custom model not from huggingface ex. @MoosaTae  LLaMa with flash attention

File: src/model/openthaigpt_pretraining_model/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    if model_config_object is None:
+        raise NotImplementedError(f"No model name: {model_config.name}")
+    model_object = MODELS.get(model_config.name, None)
+    if model_object is None:
+        raise NotImplementedError(f"No model name: {model_config.name}")
+
+    tokenizer = tokenizer_object.from_pretrained(model_config.tokenizer.pretrained)
```
Comment (new5558) [inline src/model/openthaigpt_pretraining_model/models/__init__.py:17]: Can our code support loading tokenizer from disk (tokenizer.json, spm.model etc.)

**Review by new5558 — APPROVED** [review]

**Review by ArthurMinovsky — APPROVED** [review]

File: src/model/openthaigpt_pretraining_model/models/__init__.py (hunk 2 above)
Comment (boat1603) [reply src/model/openthaigpt_pretraining_model/models/__init__.py]: Done

File: src/model/configuration_example/model/falcon-7b-from-pretrained.yaml (hunk 1 above)
Comment (boat1603) [reply src/model/configuration_example/model/falcon-7b-from-pretrained.yaml]: Done

**Pull request merged**